### ISSUE TITLE
JPEG-XL support in Fiji/ImageJ

### ISSUE THREAD

**Issue by Manpoozle** [libjxl/libjxl#2740]
**Is your feature request related to a problem? Please describe.**
ImageJ, and its helper app Fiji are used extensively in microscopy, the backbone of biology research. Researchers generally store images from microscope cameras in .tiff format, but there are a lot of little image manipulations necessary to even view these images. For instance, in epifluorescence microscopy, there will be multiple color channels that are imaged separately using a filter wheel, and contrast is not at all normalized. This happens to also apply to astrophotography imaging. Fiji allows scientists to look at these raw tiff images before putting them through an extensive image correction pipeline. Often, scientists save these images right out of Fiji, and don't need the heavy lifting of registration, background subtraction, and the like that comes with these pipelines. The amount of data that is being collected in .tiff is unsustainable without some form of fundemental technological leap in disk storage, and labs spend millions of dollars buying hundreds of petabytes of on-site storage because they cannot afford the costs of cloud solutions. In astronomy, this data is often stored on magnetic tape, and then physically trucked offsite, because the high-throughput data transfer pipelines that exist are insufficient for the volume of data they are collecting.

**Describe the solution you'd like**
Implement a JPEG-XL plugin for ImageJ and bundle with Fiji.

**Describe alternatives you've considered**
Currently, we have engineering integrating the conversion in the imaging pipeline itself. We keep an eye on the other actively-maintained, free, open source, and platform-agnostic solutions. We also have an ML-based compression that we're using concurrently. Until the adoption of JPEG-XL is more complete, and more labs are courageous enough to publish with compressed images, we still have to maintain the original .tiff images. Nothing else out there beats ImageJ through Fiji.

**Additional context**
There are a lot of imaging pipelines built on ImageJ because it can be used locally on developer and scientist MacBooks, but then also published out to local on-site and cloud infrastructure. Cloud offerings have support for containers, like Docker, but educational institutions and private labs still run almost exclusively on bare-metal and VMs. Operation teams and engineering teams in these institutions are understaffed and overwhelmed because the draw to tech companies and the compensation they can afford is beyond what they can compete with. Most labs affiliated with universities have access to a combination of old-guard engineers and bright-eyed recent college grads. Tools that are mature, free, widely adopted, flexible and have low overhead are the bread and butter of these institutions. But the exponential growth of data is forcing early-adoption of new technologies like JPEG-XL, and they are being shoe-horned in to backend pipelines without front-end support for the scientists. The hope is that this is a win/win, a single change that can both power the back-end and also power the front-end.

**Comment by jonnyawsom3** [comment]
The closest to a Java implementation currently is [JxLatte](https://github.com/Traneptora/jxlatte), which only supports decoding unfortunately. It may be worth keeping an eye on though

**Comment by Manpoozle** [comment]
Yeah, and I definitely think it could be done piecemeal. Being able to view the files is more important than being able to convert within ImageJ/Fiji. That can be left up to the imaging pipeline, which often run in language-agnostic or python-based bioinformatics workflow managers, Jupyter notebooks, etc.

**Comment by lannybroo** [comment]
There is also a [JNI wrapper included in libjxl](https://github.com/libjxl/libjxl/tree/main/tools/jni/org/jpeg/jpegxl/wrapper) that could be used for this purpose.
